code/modules/mob/living/carbon/xenomorph/abilities/predalien/predalien_powers.dm

```diff
@@ -34,124 +31,318 @@
+			predalien_smash.visible_message(SPAN_XENOWARNING("[predalien_smash]'s muscles twitch."), SPAN_XENOWARNING("Our claws twitch as we try to grab onto the target but lack the strength. Wait a moment to try again."))
+			twitch_message_cooldown = world.time + 5 SECONDS
+		return //this gives a little feedback on why your lunge didn't hit other than the lunge button going grey. Plus, it might spook marines that almost got lunged if they know why the message appeared, and extra spookiness is always good.
+
+	if (!affected_atom)
 		return
 
-	if (get_dist_sqrd(target, xeno) > 2)
-		to_chat(xeno, SPAN_XENOWARNING("[target] is too far away!"))
+	if (!isturf(predalien_smash.loc))
+		to_chat(predalien_smash, SPAN_XENOWARNING("We can't lunge from here!"))
 		return
 
-	var/mob/living/carbon/carbon = target
+	if (!predalien_smash.check_state() || predalien_smash.agility)
+		return
 
-	if (carbon.stat == DEAD)
-		to_chat(xeno, SPAN_XENOWARNING("[carbon] is dead, why would we want to touch them?"))
+	if(predalien_smash.can_not_harm(affected_atom) || !ismob(affected_atom))
+		apply_cooldown_override(click_miss_cooldown)
 		return
 
-	var/datum/behavior_delegate/predalien_base/behavior = xeno.behavior_delegate
-	if(!istype(behavior))
+	var/mob/living/carbon/carbon = affected_atom
+	if(carbon.stat == DEAD)
 		return
 
 	if (!check_and_use_plasma_owner())
 		return
 
-	ADD_TRAIT(carbon, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Devastate"))
-
-	if (ishuman(carbon))
-		var/mob/living/carbon/human/human = carbon
-		human.update_xeno_hostile_hud()
-
 	apply_cooldown()
+	..()
+	predalien_smash.visible_message(SPAN_XENOWARNING("[predalien_smash] grabs [carbon]!"), SPAN_XENOWARNING("We grab [carbon]!"))
 
-	ADD_TRAIT(xeno, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Devastate"))
-	xeno.anchored = TRUE
+	predalien_smash.throw_atom(get_step_towards(affected_atom, predalien_smash), grab_range, SPEED_FAST, predalien_smash)
 
-	if (do_after(xeno, activation_delay, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_HOSTILE))
-		xeno.visible_message(SPAN_XENOHIGHDANGER("[xeno] rips open the guts of [carbon]!"), SPAN_XENOHIGHDANGER("We rip open the guts of [carbon]!"))
-		carbon.spawn_gibs()
-		playsound(get_turf(carbon), 'sound/effects/gibbed.ogg', 75, 1)
-		carbon.apply_effect(get_xeno_stun_duration(carbon, 0.5), WEAKEN)
-		carbon.apply_armoured_damage(get_xeno_damage_slash(carbon, base_damage + damage_scale * behavior.kills), ARMOR_MELEE, BRUTE, "chest", 20)
+	if (predalien_smash.Adjacent(carbon))
+		predalien_smash.start_pulling(carbon,1)
+		if(ishuman(carbon))
+			INVOKE_ASYNC(carbon, TYPE_PROC_REF(/mob, emote), "scream")
+			carbon.apply_armoured_damage(get_xeno_damage_slash(carbon, smash_damage + smash_scale * predalienbehavior.kills), ARMOR_MELEE, BRUTE, "chest", 20)
+	else
+		predalien_smash.visible_message(SPAN_XENOWARNING("[predalien_smash]'s claws twitch."), SPAN_XENOWARNING("We coulddn't grab our target. Wait a moment to try again."))
+
+	return TRUE
```

Comment:
Is there a reason to return true to this proc?

Reply:
its for not being able to spam it